doc/configuration.rst

```diff
@@ -1087,6 +1088,60 @@ Binder links will point to these notebooks.
 See the Sphinx-Gallery `Sphinx configuration file <https://github.com/sphinx-gallery/sphinx-gallery/blob/master/doc/conf.py>`_
 for an example that uses the `public Binder server <http://mybinder.org>`_.
 
+.. _promote_jupyter_magic:
+
+Making cell magic executable in notebooks
+=========================================
+
+Often times, tutorials will include bash scripts for the user to copy/paste into
+their terminal. These scripts are not intended to be run when someone is
+building the documentation, as they will already have those dependencies in
+their environment. Hence they are normally written as code blocks inside text::
+
+  #%%
+  # Installing dependencies
+  #
+  #     .. code-block:: bash
+  #
+  #       pip install -q tensorflow
+  #       apt-get -qq install curl
+
+This works fine for the ``.py`` and ``.html`` files, but causes problems when
+rendered as an Jupyter notebook. The downloaded ``.ipynb`` file will not have
+those dependencies installed, and thus will not work without running the script.
+
+To fix this, we can set the ``promote_jupyter_magic`` flag in ``conf.py``::
+
+  sphinx_gallery_conf = {
+      ...
+      'promote_jupyter_magic': True,
+  }
+
+If this flag is ``True``, then when a Jupyter notebook is being built, any code
+block starting with a `Jupyter cell magic <https://ipython.readthedocs.io/en/stable/interactive/magics.html>`_ (e.g. ``%%bash`` or ``%%writefile``)
```

Comment:
```suggestion
block starting with `Jupyter cell magic <https://ipython.readthedocs.io/en/stable/interactive/magics.html>`_ (e.g. ``%%bash`` or ``%%writefile``)
```

Reply:
Changed to "blocks starting with Jupyter cell magics", as [the IPython docs use "magics" as the plural](https://ipython.readthedocs.io/en/stable/interactive/magics.html).